include/unifex/repeat.hpp

```diff
@@ -0,0 +1,235 @@
+      }
+    }
+  }
+
+  template<
+    typename R = Receiver,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_done), R>, int> = 0>
+  void set_done() noexcept {
+    assert(op_ != nullptr);
+    unifex::set_done(std::move(op_->receiver_));
+  }
+
+  template<
+    typename Error,
+    std::enable_if_t<std::is_invocable_v<decltype(unifex::set_error), Receiver, Error>, int> = 0>
+  void set_error(Error&& error) noexcept {
+    assert(op_ != nullptr);
+    unifex::set_error(std::move(op_->receiver_), (Error&&)error);
+  }
+
+private:
+  template<typename CPO, typename... Args>
+  friend auto tag_invoke(CPO cpo, const repeat_source_receiver& r, Args&&... args)
+      noexcept(std::is_nothrow_invocable_v<CPO, const Receiver&, Args...>)
+      -> std::invoke_result_t<CPO, const Receiver&, Args...> {
+    return std::move(cpo)(r.get_receiver(), (Args&&)args...);
+  }
+  
+  template <typename VisitFunc>
+  friend void tag_invoke(
+      tag_t<visit_continuations>,
+      const repeat_source_receiver& r,
+      VisitFunc&& func) noexcept(std::is_nothrow_invocable_v<
+                                VisitFunc&,
+                                const Receiver&>) {
+    std::invoke(func, r.get_receiver());
+  }
+
+  const Receiver& get_receiver() const noexcept {
+    assert(op_ != nullptr);   
+    return op_->receiver_;
+  }
+
+  operation* op_;
+};
+
+template<typename Source, typename Receiver>
+class repeat_operation {
+  using source_receiver = detail::repeat_source_receiver<Source, Receiver>;
+
+public:
+  template<typename Source2, typename Receiver2>
+  explicit repeat_operation(Source2&& source, Receiver2&& receiver)
+      noexcept(std::is_nothrow_constructible_v<Source, Source2> &&
+               std::is_nothrow_constructible_v<Receiver, Receiver2> &&
+               is_nothrow_connectable_v<Source&, source_receiver>)
+  : source_((Source2&&)source)
+  , receiver_((Receiver&&)receiver)
+  {
+    sourceOp_.construct_from([&] {
+        return unifex::connect(source_, source_receiver{this});
+      });
+  }
+
+  ~repeat_operation() {
+    if (isSourceOpConstructed_) {
+      sourceOp_.destruct();
+      isSourceOpConstructed_ = false;
+    }
+  }
+
+  void start() & noexcept {
+    unifex::start(sourceOp_.get());
+  }
+
+private:
+  friend repeat_source_receiver<Source, Receiver>;
+
+  using source_op_t = operation_t<Source&, repeat_source_receiver<Source, Receiver>>;
+
+  UNIFEX_NO_UNIQUE_ADDRESS Source source_;
+  UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+  bool isSourceOpConstructed_ = true;
+  manual_lifetime<source_op_t> sourceOp_;
+};
+
+} // namespace detail
+
+template<typename Source>
+class repeat_sender {
+
+public:
+  template<template<typename...> class Variant,
+           template<typename...> class Tuple>
+  using value_types = typename Source::template value_types<Variant, Tuple>;
+
+  template<template<typename...> class Variant>
+  using error_types = typename Source::template error_types<Variant>;
```

Comment:
This also needs to inject the `std::exception_ptr` type to the list of error types as it might send `std::exception_ptr` if the connecting the sender with the `repeat_source_receiver` might throw.